### ISSUE TITLE
Validating iNaturalist checklist / ParentNameUsageID invalid

### ISSUE THREAD

**Issue by sylvain-morin** [gbif/portal-feedback#5548]
Hello,

I'm trying to validate the iNaturalist checklist (https://www.inaturalist.org/taxa/inaturalist-taxonomy.dwca.zip)
and I have the following issue using the Data Validator (https://www.gbif.org/tools/data-validator)

In the Validation Report, the ParentNameUsageID is flagged as invalid for 100% of the records

https://www.gbif.org/tools/data-validator/f3a3af14-8617-4ef4-8d3b-f082a98af125

![Image](https://github.com/user-attachments/assets/dc0b10de-9c14-424a-8d3e-a62bef46377e)

I don't understand the issue since the data seems valid to me - it's a taxonID we can find in the taxonID column

Thank you.
Sylvain

**Comment by CecSve** [comment]
In the flags and issues [documentation](https://techdocs.gbif.org/en/data-use/checklist-issues-and-flags) of tech docs, the issue is described as:

> The value for the ParentNameUsageID doesn’t correspond to a valid entry in the list. Check that the parentNameUsageID points to an existing taxon entry within the checklist (parentNameUsageID should contain the value of the taxonID of the parent taxon in the same checklist

But as you state, the ID should be present in the taxonID column. I will try to investigate on our side.

**Comment by CecSve** [comment]
One issue I noticed is the taxonID `https://www.inaturalist.org/taxa/48460`. To accommodate 'Life', which is not a proper scientificName, I would suggest you instead provide `scientificName `= `incertae sedis` and provide `kingdom` = `Biota`. This would allow us to interpret that field. However, I doubt iNat has records that are research-grade at that level? If not, I would suggest not to include that entry.

**Comment by CecSve** [comment]
> One issue I noticed is the taxonID `https://www.inaturalist.org/taxa/48460`. To accommodate 'Life', which is not a proper scientificName, I would suggest you instead provide `scientificName `= `incertae sedis` and provide `kingdom` = `Biota`. This would allow us to interpret that field. However, I doubt iNat has records that are research-grade at that level? If not, I would suggest not to include that entry.

I tried to apply the suggestions and rerun the validator, and that made things worse across the board for some reason https://www.gbif.org/tools/data-validator/1159ea70-d8d7-4a24-91ce-98cb939ce674. I also changed the `metadata.eml.xml` file to `eml.xml` to make the archive validation run.

**Comment by jhnwllr** [comment]
@sylvain-morin `ParentNameUsageID` should not have `https://www.inaturalist.org/taxa/` prefix. It should just have the `id` number. I was able to run it through the validator with just the number. 

https://www.gbif.org/tools/data-validator/fedef4b3-b3ad-4b96-a7cf-5fd7a5951c8d

There are still 2 errors, but it seems to fix the majority of cases.

**Comment by sylvain-morin** [comment]
thank you @jhnwllr  for testing, but it doesn't seem to match the DarwinCore definition of  https://dwc.tdwg.org/terms/#dwc:parentNameUsageID

it says:

> This term should be used for accepted names to refer to the **dwc:taxonID** of a dwc:Taxon record

here is my meta.xml:

```
    <id index="0"/>
    <field index="1" term="http://rs.tdwg.org/dwc/terms/taxonID"/>
    <field index="2" term="http://purl.org/dc/terms/identifier"/>
    <field index="3" term="http://rs.tdwg.org/dwc/terms/parentNameUsageID"/>

```

and my data:

![Image](https://github.com/user-attachments/assets/fc21e133-633f-4fa2-bbc2-dd3023a4be06)

so IMO, we should rather keep using the `taxonID` format, with `https://www.inaturalist.org/taxa/` prefix, than the `id` format (integer)

**Comment by jhnwllr** [comment]
@sylvain-morin it is hard to diagnose validator issues unfortunately. I am not 100% my "fix" is correct.

**Comment by sylvain-morin** [comment]
the only code ref I can find is :
https://github.com/gbif/checklistbank/blob/master/checklistbank-cli/src/main/java/org/gbif/checklistbank/cli/normalizer/Normalizer.java#L734

this error is send when the parent is not found, using `nodeByTaxonId(id)`

defined here:
https://github.com/gbif/checklistbank/blob/master/checklistbank-cli/src/main/java/org/gbif/checklistbank/neo/ImportDb.java#L58

which is using TAXON_ID = DwcTerm.taxonID.simpleName()
https://github.com/gbif/checklistbank/blob/master/checklistbank-cli/src/main/java/org/gbif/checklistbank/neo/NeoProperties.java#L28

so, again, the column TaxonID and not ID, I would say...

**Comment by CecSve** [comment]
You are right @sylvain-morin. Removing the ID column from the meta.xml file and core table almost resolves the issue with parentNameUsageID: https://www.gbif.org/tools/data-validator/cc9a397f-eebd-499b-98cc-a9c59719b1d5 - I assume the remaining three issues are fixable. You technically only need to resolve the red flags and issues for the archive to ingest properly.
